app/models/piece.rb

```diff
@@ -12,5 +12,56 @@ class Piece < ActiveRecord::Base
 	def self.types
 		%w(Pawn Knight Bishop Rook Queen King)
 	end
+
+	# Determine if there are obstructions "between" two chess pieces
+	# to validate a piece can succesfully be moved.  Accepts x and y coordinates 
+	# for both pieces being compared.  Returns false if no obstruction exists, true if obstruction exists.
+	def is_obstructed?(start_point, end_point)
```

Comment:
nevermind this comment.  I re-read the code and it looks correct, you should probably still write a test or two to verify the behavior and guard against regressions but I think this is good.